ruby/ql/lib/codeql/ruby/frameworks/http_clients/Httparty.qll

```diff
@@ -39,7 +39,7 @@ class HttpartyRequest extends HTTP::Client::Request::Range {
     exists(DataFlow::Node r | r = requestNode.getAMethodCall("body") | result = r)
     or
     // Otherwise, treat the response as the response body.
-    not exists(DataFlow::Node r | r = requestNode.getAMethodCall("body")) and
```

Comment:
I wonder if the line
```
exists(DataFlow::Node r | r = requestNode.getAMethodCall("body") | result = r)
```
could also be simplified to
```
result = requestNode.getAMethodCall("body")
```
if the query was extended slightly? (We know that `result` has type `DataFlow::Node` in this case.)

Reply:
It could, and originally my query did just that (see [the discussion in the PR](https://github.com/github/codeql/pull/7472)).  
But that had too many FPs, so I dropped it.